### PR TITLE
Move ScalarAggregationToJoinRewriter to TransformCorrelatedScalarAggregationToJoin

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java [hunk 1]
```diff
@@ -114,4 +135,157 @@ public Result apply(LateralJoinNode lateralJoinNode, Captures captures, Context
                 .recurseOnlyWhen(isInstanceOfAny(ProjectNode.class, EnforceSingleRowNode.class))
                 .findFirst();
     }
+
+    private static class Rewriter
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java]: final

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java [hunk 2]
```diff
@@ -16,23 +16,44 @@
 import com.facebook.presto.matching.Captures;
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java]: this conflicts with https://github.com/prestodb/presto/pull/8435

**Comment by sopel39** [comment]
Why so?

**Comment by kokosing** [comment]
As far I recall that PR is using this class from two different rules.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java [hunk 3]
```diff
@@ -114,4 +135,157 @@ public Result apply(LateralJoinNode lateralJoinNode, Captures captures, Context
...
+    {
+        private static final QualifiedName COUNT = QualifiedName.of("count");
+
+        private final FunctionRegistry functionRegistry;
+        private final SymbolAllocator symbolAllocator;
+        private final PlanNodeIdAllocator idAllocator;
+        private final Lookup lookup;
+        private final PlanNodeDecorrelator planNodeDecorrelator;
+
+        public Rewriter(FunctionRegistry functionRegistry, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator, Lookup lookup)
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java]: not `public`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ScalarAggregationToJoinRewriter.java [hunk 4]
```diff
@@ -1,204 +0,0 @@
...
-import java.util.Optional;
-import java.util.Set;
-
-import static com.facebook.presto.sql.analyzer.TypeSignatureProvider.fromTypeSignatures;
-import static com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
-import static com.facebook.presto.sql.tree.BooleanLiteral.TRUE_LITERAL;
-import static com.google.common.collect.ImmutableList.toImmutableList;
-import static java.util.Objects.requireNonNull;
-
-// TODO: move this class to TransformCorrelatedScalarAggregationToJoin when old optimizer is gone
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ScalarAggregationToJoinRewriter.java]: Is the removal of this line the only change here?
If so, could you add sth like that to the cmt msg:
```
The `ScalarAggregationToJoinRewriter` class was used by old optimizer and iterative rule and the old optimizer has been removed.
```

**Pull request closed without merging**